homeassistant/components/bluetooth/scanner.py

```diff
@@ -385,15 +385,11 @@ async def _async_reset_adapter(self) -> None:
 
     async def async_stop(self) -> None:
         """Stop bluetooth scanner."""
-        async with self._start_stop_lock:
-            await self._async_stop()
-
-    async def _async_stop(self) -> None:
-        """Cancel watchdog and bluetooth discovery under the lock."""
         if self._cancel_watchdog:
             self._cancel_watchdog()
             self._cancel_watchdog = None
-        await self._async_stop_scanner()
+        async with self._start_stop_lock:
```

Comment:
I don't understand the PR description. Canceling the watchdog can't hold the lock as the cancel function is not a coroutine function.

Reply:
When the host system is in a broken state and the reset is timing out, we want to prevent the watchdog from firing again since it holds the start stop lock which would block the stop call from happening.

Reply:
We should probably add a check in the watchdog as well to see if it's still holding the lock from the last attempt since and not try to reset again since it means the system is in such a bad state it's taking longer to do the reset than the watchdog interval. 

I've only seen that happen when the host system runs out of ram from something unrelated on my rpi3b or there is a kernel crash as it gets dbus and Bluetoothd in an unrecoverable state that makes the Bluetooth management socket stall forever

Reply:
https://github.com/home-assistant/core/pull/80679